nordicsemi/dfu/dfu_transport_ble.py

```diff
@@ -531,37 +532,39 @@ def try_to_recover():
                 self.__stream_data(data=init_packet)
                 self.__execute()
             except ValidationException as error:
-                logger.critical(f"BLE: Error occurred during init packet send at attempt {r + 1}: {error}")
+                logger.critical(f"BLE: ValidationException Error occurred during init packet send at "
+                                f"attempt {r + 1}: {error}")
                 continue
             break
         else:
             raise NordicSemiException("Failed to send init packet")
 
     def send_firmware(self, firmware):
         def try_to_recover():
+            logger.debug("BLE: Trying to recover...")
             if response['offset'] == 0:
-                # Nothing to recover
+                logger.debug("BLE: Nothing to recover")
                 return
 
             expected_crc = binascii.crc32(firmware[:response['offset']]) & 0xFFFFFFFF
             remainder    = response['offset'] % response['max_size']
 
             if expected_crc != response['crc']:
-                # Invalid CRC. Remove corrupted data.
+                logger.debug("BLE: Invalid CRC. Remove corrupted data.")
```

Comment:
I know, those are debug/internal logs, but less confusing would be `-ing` form, like `Removing` or `Sending` (at line 559) - as that would inform what the script is doing instead of suggesting an action to be taken by an user.